### PR TITLE
Add test coverage for gnm_common

### PR DESCRIPTION
**Pull request by manumishra12**
Adds `gnm_common_test.py`. `gnm_common.py` holds the backend-agnostic math
shared by all four GNM backends but had no dedicated test (it was only
exercised indirectly through the per-backend tests).

The new suite (25 tests) covers:

- **Array helpers**: `take`, `eye`, `reshape_with_batch_dims`,
  `zeros_with_batch_dims`.
- **axis_angle_to_rotation_matrix**: zero -> identity, rotation about each
  principal axis leaves that axis invariant, the 90 degrees-about-z reference
  matrix, orthonormality and unit determinant on random inputs, and batch-shape
  preservation.
- **joint_transforms_world**: rest pose places joints at their bind positions,
  a global translation shifts the chain, and a root rotation propagates to the
  child joint.
- **linear_blend_skinning**: rest pose is the identity, a global translation
  translates the vertices, and batch shapes are preserved.
- **vertex_positions_bind_pose / joint_positions_bind_pose**: `None`
  parameters return the template, and identity/expression bases are applied
  correctly.
- **compute_pose_correctives**: `None` inputs and zero rotations produce zero
  correctives, and batch shapes are preserved.

Follows the existing `absltest` style used across the package. All tests pass
locally with the NumPy backend.

### CODE REVIEW

File: gnm/shape/gnm_common_test.py [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+
+from absl.testing import absltest
+from absl.testing import parameterized
+from gnm.shape import gnm_common
+
+
+class ArrayHelpersTest(parameterized.TestCase):
+  """Tests for the backend-agnostic array helpers."""
+
+  def test_take_selects_along_axis(self) -> None:
```
Comment (bednarikjan) [inline gnm/shape/gnm_common_test.py]: Please remove the return type annotation from all the `test_*` functions. This is to stay consistent with our internal codebase.

File: gnm/shape/gnm_common_test.py [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+from __future__ import annotations
+
+import numpy as np
+
+from absl.testing import absltest
+from absl.testing import parameterized
+from gnm.shape import gnm_common
+
+
+class ArrayHelpersTest(parameterized.TestCase):
```
Comment (bednarikjan) [inline gnm/shape/gnm_common_test.py:36]: For all the tests, please use the `parameterized` to generate a given test for all the backends supported by xnp (NumPy, jax, TF, pytorch).

For the functions which take `xnp` as an input argument, you can iterate over the backends directly (take, eye), for the other functions you can iterate over input arrays built by the individual backends from which the current xnp is extracted at runtime within the functions under test (reshape_with_batch_dims, zeros_with_batch_dims).

This will help the tests catch any potential bugs specific to a given backend.

File: gnm/shape/gnm_common_test.py [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+    reference = np.ones((4, 5, 3), dtype=np.float32)
+    zeros = gnm_common.zeros_with_batch_dims(
+        reference, num_reference_non_batch_dims=2, suffix_shape=(4, 4),
+        dtype=np.float32,
+    )
+    self.assertEqual(zeros.shape, (4, 4, 4))
+    np.testing.assert_array_equal(zeros, np.zeros((4, 4, 4)))
+
+
+class AxisAngleToRotationMatrixTest(parameterized.TestCase):
```
Comment (bednarikjan) [inline gnm/shape/gnm_common_test.py:87]: As in the comment about ArrayHelperTest, please use the `parameterized` to iterate as well over the supported backends. Fo the actual asserts, for simplicity one could always convert the values to np.ndarray.

File: gnm/shape/gnm_common_test.py [hunk 4]
```diff
@@ -0,0 +1,264 @@
...
+    )
+    np.testing.assert_allclose(np.linalg.det(matrices), np.ones(8), atol=1e-6)
+
+  def test_preserves_batch_shape(self) -> None:
+    axis_angle = np.zeros((2, 4, 3))
+    matrices = gnm_common.axis_angle_to_rotation_matrix(axis_angle)
+    self.assertEqual(matrices.shape, (2, 4, 3, 3))
+
+
+class JointTransformsWorldTest(parameterized.TestCase):
```
Comment (bednarikjan) [inline gnm/shape/gnm_common_test.py:142]: As in the comment about ArrayHelperTest, please use the `parameterized` to iterate as well over the supported backends. Fo the actual asserts, for simplicity one could always convert the values to np.ndarray.

File: gnm/shape/gnm_common_test.py [hunk 5]
```diff
@@ -0,0 +1,264 @@
...
+    transforms = gnm_common.joint_transforms_world(
+        self.joints, rotations, np.zeros(3), self.parents
+    )
+    np.testing.assert_allclose(
+        self._translations(transforms)[1], np.array([0.0, 1.0, 0.0]),
+        atol=1e-6,
+    )
+
+
+class LinearBlendSkinningTest(parameterized.TestCase):
```
Comment (bednarikjan) [inline gnm/shape/gnm_common_test.py:206]: As in the comment about ArrayHelperTest, please use the `parameterized` to iterate as well over the supported backends. Fo the actual asserts, for simplicity one could always convert the values to np.ndarray.

File: gnm/shape/gnm_common_test.py [hunk 6]
```diff
@@ -0,0 +1,264 @@
...
+  def test_preserves_batch_shape(self) -> None:
+    batch_vertices = np.broadcast_to(self.vertices, (4, 3, 3))
+    posed = gnm_common.linear_blend_skinning(
+        batch_vertices, np.broadcast_to(self.joints, (4, 2, 3)),
+        np.zeros((4, 2, 3)), np.zeros((4, 3)), self.weights, self.parents,
+    )
+    self.assertEqual(posed.shape, (4, 3, 3))
+
+
+class BindPoseTest(parameterized.TestCase):
```
Comment (bednarikjan) [inline gnm/shape/gnm_common_test.py:264]: As in the comment about ArrayHelperTest, please use the `parameterized` to iterate as well over the supported backends. Fo the actual asserts, for simplicity one could always convert the values to np.ndarray.

File: gnm/shape/gnm_common_test.py [hunk 7]
```diff
@@ -0,0 +1,264 @@
...
+    joint_basis = np.array([[[1.0, 0.0, 0.0], [0.0, 1.0, 0.0]]])  # (I=1, J=2, 3)
+    result = gnm_common.joint_positions_bind_pose(
+        np.array([2.0]), template, joint_basis
+    )
+    np.testing.assert_allclose(
+        result, np.array([[2.0, 0.0, 0.0], [0.0, 2.0, 0.0]]), atol=1e-6
+    )
+
+
+class PoseCorrectivesTest(parameterized.TestCase):
```
Comment (bednarikjan) [inline gnm/shape/gnm_common_test.py:324]: As in the comment about ArrayHelperTest, please use the `parameterized` to iterate as well over the supported backends. Fo the actual asserts, for simplicity one could always convert the values to np.ndarray.

**Comment by bednarikjan** [comment]
@manumishra12 Thanks for the PR! Besides the comments above, please see also the output of the linter CI.

**Comment by manumishra12** [comment]
Thanks for the review! I've pushed an update addressing all the comments:

- **Return type annotations removed** — dropped the `-> None` annotations from every `test_*` function (and from the two `setUp` methods) to stay consistent with the internal codebase.

- **Parameterized over all backends** — every test now runs against all four xnp backends via a shared `_XNP_BACKENDS` list decorated with `@parameterized.named_parameters`:

  ```python
  _XNP_BACKENDS = (
      ('numpy', np),
      ('jax', enp.lazy.jnp),
      ('tensorflow', enp.lazy.tnp),
      ('pytorch', enp.lazy.torch),
  )
  ```

  For the functions that take an `xnp` argument (`take`, `eye`) the backend module is passed through directly. For the rest, the inputs are built with each backend so the current `xnp` is extracted at runtime inside the function under test. Assertions convert the results to `np.ndarray` for simplicity. This gives 23 test methods x 4 backends = 92 test cases.

- **Linter** — fixed the `line-too-long` failure; `pylint gnm/` is now clean.

All 92 tests pass locally against NumPy, JAX, TensorFlow, and PyTorch. Let me know if you'd like any further changes.

**Review by bednarikjan — APPROVED** [review]
